src/blas/KokkosBlas3_gemm.hpp

```diff
@@ -256,6 +256,105 @@ void gemm(const char transA[], const char transB[],
   gemm(space, transA, transB, alpha, A, B, beta, C);
 }
 
+/********************* BEGIN functor-level routines *********************/
+
+///
+/// Serial Gemm
+///
+
+template <typename ArgTransA, typename ArgTransB, typename ArgAlgo>
+struct SerialGemm {
+  template <typename ScalarType, typename AViewType, typename BViewType,
+            typename CViewType>
+  KOKKOS_INLINE_FUNCTION static int invoke(const ScalarType alpha,
+                                           const AViewType& A,
+                                           const BViewType& B,
+                                           const ScalarType beta,
+                                           const CViewType& C);
+};
+
+///
+/// Team Impl
+/// =========
+
+template <typename ArgTransA, typename ArgTransB, typename ArgAlgo>
+struct TeamGemm {
+  template <typename MemberType, typename ScalarType, typename AViewType,
+            typename BViewType, typename CViewType>
+  KOKKOS_INLINE_FUNCTION static int invoke(
+      const MemberType& member, const ScalarType alpha, const AViewType& A,
+      const BViewType& B, const ScalarType beta, const CViewType& C);
+};
+
+///
+/// TeamVector Impl
+/// =========
+
+template <typename ArgTransA, typename ArgTransB, typename ArgAlgo>
+struct TeamVectorGemm {
+  template <typename MemberType, typename ScalarType, typename AViewType,
+            typename BViewType, typename CViewType>
+  KOKKOS_INLINE_FUNCTION static int invoke(
+      const MemberType& member, const ScalarType alpha, const AViewType& A,
+      const BViewType& B, const ScalarType beta, const CViewType& C);
+};
+
+///
+/// Selective Interface
+///
+template <typename ArgTransA, typename ArgTransB, typename ArgMode,
+          typename ArgAlgo = Algo::Gemm::Default>
+struct Gemm {
```

Comment:
@lucbv Should this interface be placed in `KokkosBlas::Experimental` namespace ? (it wasn't in `KokkosBatched`)